RevenueCatUI/PaywallView.swift

```diff
@@ -106,9 +138,14 @@ public struct PaywallView: View {
     ) {
         self._introEligibility = .init(wrappedValue: introEligibility ?? .default())
         self._purchaseHandler = .init(wrappedValue: purchaseHandler)
-        self._offering = .init(
-            initialValue: offering ?? Self.loadCachedCurrentOfferingIfPossible()
-        )
+        self._offeringSelection = .init(wrappedValue: offeringSelection)
+        var initialOffering: Offering?
+        if let offering {
+            initialOffering = offering
+        } else if offeringSelection == nil {
+            initialOffering = Self.loadCachedCurrentOfferingIfPossible()
```

Comment:
can we offeringSelection(Self.cachedOfferings) here?

Reply:
I think we don't want to do it here, because this way we may never get the specific offering that we want to choose

Reply:
Updated to use cached offerings